Sources/ArcGISToolkit/Components/Augmented Reality/WorldScaleSceneView.swift

```diff
@@ -262,6 +262,31 @@ private extension WorldScaleSceneView {
     }
 }
 
+public extension WorldScaleSceneView {
+    /// Determines the scene point for the given screen point.
+    /// - Parameter screenPoint: The point in screen's coordinate space.
+    /// - Returns: The scene point corresponding to screen point.
+    private func arScreenToLocation(screenPoint: CGPoint) -> Point? {
+        // Use the `raycast` method to get the matrix of `screenPoint`.
+        guard let localOffsetMatrix = arViewProxy.raycast(from: screenPoint, allowing: .estimatedPlane) else { return nil }
+        let originTransformationMatrix = cameraController.originCamera.transformationMatrix
+        let scenePointMatrix = originTransformationMatrix.adding(localOffsetMatrix)
+        // Create a camera from transformationMatrix and return its location.
+        return Camera(transformationMatrix: scenePointMatrix).location
+    }
+    
+    /// Sets a closure to perform when a single tap occurs on the view.
+    func onSingleTapGesture(
+        perform action: @escaping (_ screenPoint: CGPoint, _ scenePoint: Point?) -> Void
```

Comment:
in what cases will this `scenePoint` be `nil`?

Reply:
When `TransformationMatrix` is `nil`, which can be caused by a failed [raycast](https://developer.apple.com/documentation/arkit/arscnview/3194595-raycastquery). Or in other words, if raycast query result is `nil`, the scene point is `nil`.

Reply:
I see that. I was wondering in what cases the recast query result is nil. Thank you

Reply:
I added some explanation to the docstrings.